internal/core/metadata/v2/controller/http/device_test.go

```diff
@@ -160,3 +162,115 @@ func TestAddDevice(t *testing.T) {
 		})
 	}
 }
+
+func TestCheckDeviceExistsById(t *testing.T) {
+	testId := ExampleUUID
+	notFoundId := "notFoundId"
+	emptyId := ""
+
+	dic := mockDic()
+	dbClientMock := &dbMock.DBClient{}
+	dbClientMock.On("CheckDeviceExistsById", testId).Return(true, nil)
+	dbClientMock.On("CheckDeviceExistsById", notFoundId).Return(false, nil)
+	dic.Update(di.ServiceConstructorMap{
+		v2MetadataContainer.DBClientInterfaceName: func(get di.Get) interface{} {
+			return dbClientMock
+		},
+	})
+
+	controller := NewDeviceController(dic)
+	assert.NotNil(t, controller)
+
+	tests := []struct {
+		name               string
+		deviceId           string
+		errorExpected      bool
+		expectedStatusCode int
+	}{
+		{"Valid - check device by id", testId, false, http.StatusOK},
+		{"Invalid - id parameter is empty", emptyId, true, http.StatusBadRequest},
+		{"Invalid - device not found by id", notFoundId, false, http.StatusNotFound},
```

Comment:
Add one for bad id if you add check that id is proper UUID

Reply:
Added bad test case for invalid UUID.